File: Support Vector Machines/Python codes/salary.py
# -*- coding: utf-8 -*-
"""
Created on Thu Oct 24 13:02:59 2019

@author: sanjay
"""

import pandas as pd
import numpy as np
salary_train=pd.read_csv('file:///G:/data/assignements/SVM/SalaryData_Train.csv')
salary_test=pd.read_csv('file:///G:/data/assignements/SVM/SalaryData_Test.csv')
string_columns=["workclass","education","maritalstatus","occupation","relationship","race","sex","native"]
for i in string_columns:
    salary_train[i] = lb.fit_transform(salary_train[i])
    salary_test[i] = lb.fit_transform(salary_test[i])
from sklearn.preprocessing import LabelEncoder
lb=LabelEncoder()
salary_test["Salary"]=lb.fit_transform(salary_test["Salary"])
salary_train["Salary"]=lb.fit_transform(salary_train["Salary"])

from sklearn.svm import SVC
train_x=salary_train.iloc[:,0:13]
train_y=salary_train.iloc[:,13]
test_x=salary_test.iloc[:,0:13]
test_y=salary_test.iloc[:,13]

#linear
model_linear=SVC(kernel="linear")
model_linear.fit(train_x,train_y)
pred_test_linear=model_linear.predict(test_x)

np.mean(pred_test_linear==test_y) # Accuracy = 82.03

# Kernel = poly
model_poly = SVC(kernel = "poly")
model_poly.fit(train_X,train_y)
pred_test_poly = model_poly.predict(test_X)

np.mean(pred_test_poly==test_y) # Accuracy = 80.42

# kernel = rbf
model_rbf = SVC(kernel = "rbf")
model_rbf.fit(train_X,train_y)
pred_test_rbf = model_rbf.predict(test_X)

np.mean(pred_test_rbf==test_y) # Accuracy = 85.44


# kernel = sigmoid
model_sig = SVC(kernel = "sigmoid")
model_sig.fit(train_X,train_y)
pred_test_sig = model_sig.predict(test_X)

np.mean(pred_test_sig==test_y) # Accuracy = 75.43

'''
# Kernel rbf model is giving high accuraccy.

File: Naive Bayes/R codes/salary.R
train <- SalaryData_Train 
test <- SalaryData_Test  
str(train)
str(train)
train$educationno<-as.factor(train$educationno)
class(train)

str(test)
test$educationno<-as.factor(test$educationno)

##visualization
ggplot(data=train,aes(x=train$Salary, y = train$age, fill = train$Salary)) +
  geom_boxplot() +
  ggtitle("Box Plot")

## age from 25- 45 has <50k salary and age 35 and above has salary>50 k 

plot(train$workclass,train$Salary)
plot(train$education,train$Salary)
plot(train$maritalstatus,train$Salary)
plot(train$educationno,train$Salary)
plot(train$occupation,train$Salary)
plot(train$relationship,train$Salary)
plot(train$race,train$Salary)
plot(train$sex,train$Salary)
plot(train$hoursperweek,train$Salary)

##model 
Model <- naiveBayes(train$Salary ~ ., data = train)
Model



Model_pred <- predict(Model,test)
mean(Model_pred==test$Salary)


confusionMatrix(Model_pred,test$Salary)
accuracy= mean(Model_pred==test$Salary)
